calcite-adapter/src/main/java/software/amazon/documentdb/jdbc/calcite/adapter/DocumentDbRules.java

```diff
@@ -373,42 +379,47 @@ protected RexToMongoTranslator(final JavaTypeFactory typeFactory,
             switch (literal.getType().getSqlTypeName()) {
                 case DOUBLE:
                 case DECIMAL:
-                    return new Operand("{\"$numberDouble\": \"" + literal.getValueAs(Double.class) + "\"}");
+                    final String doubleFormat = "{\"$numberDouble\": \"" + literal.getValueAs(Double.class) + "\"}";
+                    return new Operand(doubleFormat, doubleFormat, true);
```

Comment:
This format for literals will work in either syntax. Any literal is fine on the right of a query operator so isQuerySyntax is true here. Same for other types below.